src/msg/async/net_handler.cc

```diff
@@ -200,15 +201,22 @@ int NetHandler::generic_connect(const entity_addr_t& addr, const entity_addr_t &
 
   ret = ::connect(s, addr.get_sockaddr(), addr.get_sockaddr_len());
   if (ret < 0) {
-    ret = errno;
-    if (errno == EINPROGRESS && nonblock)
+    ret = ceph_sock_errno();
+    // TODO: define some macros to check those states.
```

Comment:
Nit: I think that would be a cleaner approach than having all these special `WSA...` error codes sprinkled throughout.